docker/config/constraints.txt

```diff
@@ -9,7 +9,7 @@ Flask-Caching==1.10.1
 Flask-JWT-Extended==3.25.1
 Flask-Login==0.4.1
 Flask-OAuthlib==0.9.5
-Flask-OpenID==1.2.5
+Flask-OpenID==1.3.0
```

Comment:
Thanks for the PR and sorry for the delay in review. There is some internal transfer of ownership we are working through :)

---
It appears the issue here is that setuptools has dropped support for `use_2to3` (see [release notes here](https://setuptools.pypa.io/en/latest/history.html#v58-0-0)) this is why you're seeing the failure in #72:
```
  error in Flask-OpenID setup command: use_2to3 is invalid.
```
From the logs of `build-image` you can see that we initially install v49 of `setuptools`:
```
---> Package python3-setuptools.noarch 0:49.1.3-1.amzn2.0.2 will be installed
```
But as of Dec 29 `psycopg2` released a new version of it's package (2.9.3 [seen here](https://pypi.org/project/psycopg2/#history)) which builds with a newer version of `setuptools`, seen in the logs:
```
...
Collecting setuptools<59.7.0,>=59.1.1
  Downloading setuptools-59.6.0-py3-none-any.whl (952 kB)
...
  Attempting uninstall: setuptools
    Found existing installation: setuptools 49.1.3
    Uninstalling setuptools-49.1.3:
      Successfully uninstalled setuptools-49.1.3
...
Successfully installed amqp-5.0.9 billiard-3.6.4.0 boto3-1.20.42 ... **setuptools-59.6.0** ...
...
Successfully installed psycopg2-2.9.3
```
This version of `setuptools` causes `Flask-OpenID` to fail as you've seen.

But I'm not sure the correct fix is updating the constraints file. That file is vended by the Airflow project itself as "known good" versions of dependencies. So I'm not inclined to patch this file on our end. I'd prefer to install a workable version of setuptools in our bootstrap.sh file, which is code we "own" in this repository and we can use it to setup the image as needed.
Something like this:
```python
# setuptools dropped support for use_2to3 in v58+ and psycopg2 will install the latest v59+ version
pip3 install $PIP_OPTION "setuptools<=57.*"                                                        
```
Added on the line **above** the airflow install and **below** the psycopg2 install in bootstrap.sh:
https://github.com/aws/aws-mwaa-local-runner/blob/6e5925bafba68d803e807efc98d53233d80796ad/docker/script/bootstrap.sh#L17-L21

What do you think? I'd also like to hear what @dhegberg thinks as well.

Reply:
What you said makes complete sense - I should have done more research on this. I'll make the change as you suggested and push it if @dhegberg agrees with you. Thank you very much!

Reply:
I agree with the approach of leaving the constraints file untouched an making the changes in bootstrap.sh.